Release v25030084

CertainLach opened the pull request:
Не мержить - напоминаю что master теперь обновляем через fast-forward

Гитхаб в своём UI такое не умеет, а потому только через консоль

Когда CI пройдёт и всё хорошо - кто-то с правом обновления мастера (Например я) сделает

```
git switch master
git merge --ff-only release-v25030084
git push origin HEAD
```
Либо
```
# master уже должен быть отслеживаемой веткой: jj bookmark track master@origin
jj bookmark set master -r release-v25030084
jj git push
```

mrshiposha committed:
fix: fast track proposal script

CertainLach committed:
build: update flake

CertainLach committed:
build: update Cargo.lock

CertainLach committed:
feat: configurable CheckAssociatedRelayNumber

Commit:
switch project to deno

Commit:
refactor typescript tests

Commit:
test fixes

Commit:
increase `MaxMemoryPages` for relay

Commit:
update imports

Commit:
Revert debug code

Commit:
switch CI  to deno

Commit:
switch codestyle check to deno

Commit:
fix deno warnings

Commit:
fix for CI

Commit:
fix unref

Commit:
fix CI

Commit:
fix imports

Commit:
fix imports

Commit:
fix imports

Commit:
fix skip collators

Commit:
fix not awaited Promise

Commit:
fix script export

Commit:
finish test refactoring

Commit:
fix install

Commit:
wait more for eth events

Commit:
allow-net for benchmark tests

Commit:
allow-read for benchmark tests

Commit:
allow-write for benchmark tests

mrshiposha committed:
feat: prepare for ahm

mrshiposha committed:
chore: bench foreign-assets

mrshiposha committed:
chore: use foreign-assets weights

fairax committed:
add `--allow-net` to createHrmp

Commit:
fix polkadot-types script

fairax committed:
deno permisions for `Reconcile: runtime updating`

mrshiposha committed:
test: exchange DOT with its reserve location

mrshiposha committed:
feat: impl XcmDryRunApi

mrshiposha committed:
fix: sendDot promises

fairax committed:
update opal runtime version to 10180083

fairax committed:
update quartz runtime version to 10180083

fairax committed:
update unique runtime version to 10180083

mrshiposha committed:
fix(tests): script permissions, sloppy-imports because of pjs types

mrshiposha committed:
fix(eth test util): remove repeating, restore possibly no-op await

mrshiposha committed:
chore: remove unneeded file

mrshiposha committed:
build: update ethers

mrshiposha committed:
build: update flake inputs, add master pkgs because of deno

mrshiposha committed:
fix: flake.nix

mrshiposha committed:
fix: maintenance script export

mrshiposha committed:
fix: remove unneeded --sloppy-imports

mrshiposha committed:
fix: xcm router

mrshiposha committed:
chore: deno -A for net+rw for now, needs proper refactor later

mrshiposha committed:
feat: add xcm runtime apis

CertainLach committed:
chore: add third-party bootnodes

CertainLach committed:
Merge pull request #1118 from UniqueNetwork/feature/third-party-boot-nodes

fairax committed:
Update pallets/foreign-assets/src/lib.rs

Co-authored-by: Daniel Shiposha <mrshiposha@gmail.com>

Commit:
move oracles to foreign-assets pallet

Commit:
add support for sponsoring

Commit:
add ocw tests to CI

Commit:
Support assets for fee

Commit:
temp

Commit:
fix linter errors

Commit:
fix clippy warnings

Commit:
add collator-selection to unique as requirement for OnChargeAssetTransaction

Commit:
remove duplicating events when fee is paid

Commit:
fix ci

Commit:
switch ocw tests to quartz

Commit:
add configuration for ocw

Commit:
fix typo

Commit:
fix clippy warnings

Commit:
transfer fee to treasury, use LocationV5 for assets

Commit:
update polkadot-types

Commit:
add new pallets to requiredPallets test

Commit:
fix code review

Commit:
fix clippy warnings

mrshiposha committed:
feat: upgrade token properties size limit

mrshiposha committed:
fix: token properties related benchmarks

ALFecki committed:
Added test for setting property size limit with ETH

mrshiposha committed:
test: token prop limit

ALFecki committed:
Added eth tests for property limits

mrshiposha committed:
feat: add token prop limit extrinsics to playgrounds

mrshiposha committed:
feat: max limit + collection set token limit support

mrshiposha committed:
chore: add renaming TODO

mrshiposha committed:
fix: token properties test

mrshiposha committed:
fix: create token weight

mrshiposha committed:
fix: consumed space + upgrade prop size limit

Commit:
fix clippy warnings

Commit:
fix tests

PraetorP committed:
test(xcm): fix XCM tests for Unique

Commit:
move PropertySizeLimitUpgradePrice to constants

PraetorP committed:
fix(xcm): update message handling for Acala and Polkadot asset hub in XCM tests

Commit:
fix tests

Commit:
fix performance test for property size

PraetorP committed:
fix(xcm): update transfer result type to ITransactionResult in XCM tests

Commit:
fix deno_version in .env file

Commit:
fix baedeker_version in .env

CertainLach committed:
build: seal release v25030084

CertainLach committed:
build: missing cargo lock update

CertainLach committed:
test: post deno typing fix

PraetorP committed:
test(xcm): reduce flakiness by reordering send/await (receiver-first)

- Mitigates race where balanceBefore could be captured after deposit if Relay arrives fast
- Start receiver-side await before sender-side send to improve timing guarantees
- Lowers probability of flakes in “DOT transfer from Unique suspended” and related tests (not a full elimination)

Commit:
fix unit tests

PraetorP committed:
fix: update call index for remove_oracle_member function

Commit:
feat: update to polkadot 2509

Commit:
set chainQL params in baedeker up.sh

Commit:
disable removing container for xcm

Commit:
set chainQL params for main-process-update workflow

Commit:
fix: weight reclaim, blocks not pruducing

Commit:
disable builds for 2 branches on CI

Commit:
ci: fixes

Commit:
style: satisfy clippy

Commit:
cargo fmt

CertainLach committed:
fix: clamp space_limit in with_space_limit_ref

CertainLach committed:
ci: fix jemalloc build issue on Nix

Ref: https://github.com/tikv/jemallocator/issues/108

CertainLach committed:
feat: set relay_parent_offset to 1

CertainLach committed:
fix(pallet-foregin-assets): call index collision

CertainLach commented:
Для evm-coder: в макросе для кодогенератора пропущен импорт trait Default, из-за чего сейчас он сломан:
<img width="1331" height="296" alt="image" src="https://github.com/user-attachments/assets/35961fdf-c674-4c02-bbdf-3ae2c261819e" />

Надо поправить и перегенерировать стабы

CertainLach commented:
Есть веса с непрогнанными бенчмарками, надо будет их прогнать

Commit:
fix: clippy warnings

Commit:
update polkadot types

CertainLach commented on js-packages/types/augment-api-errors.ts:
```diff
@@ -265,7 +265,7 @@ declare module '@polkadot/api-base/types/errors' {
        * Insufficient funds to perform an action
        **/
-      NotSufficientFunds: AugmentedError<ApiType>;
+      NotSufficientFounds: AugmentedError<ApiType>;
```
Мне кажется ты типы откатил, на develop оно NotSufficientFunds

fairax replied:
Похоже у меня не обновились бинарники когда я сделал cargo build

Commit:
fix update polkadot types

This reverts commit f65c36e337eb171749e4178539cf43fd11eb4bc1.

Commit:
ci: cargo fmt

Commit:
up mainnet branch version for CI

Commit:
ci: deno lint

Commit:
Revert "feat: set relay_parent_offset to 1"

This reverts commit a235e99d2512a418b3265885595c75e54785bd3f.

mrshiposha committed:
fix: per-asset foreign-assets coeff

mrshiposha committed:
fix: benchmarks

mrshiposha committed:
fix(foreign-assets): feature not(governance) manager origin

mrshiposha committed:
feat: add SupremeTrio, use for foreign assets mgmt

mrshiposha committed:
chore: fmt

mrshiposha committed:
fix: add -lxcm=trace to all bdk nodes

mrshiposha committed:
build: benchmark weights

NOTE: the scheduler must be monitored in the release CI
as the temporary fix for its weights is gone now.
It should work properly but still must be watched closely before release

mrshiposha committed:
chore: calibrate weight coeffs

Commit:
enable clean docker artifacts for xcm wf

Commit:
fix CI

Commit:
fix CI

Commit:
Fix CI

Commit:
fix CI

Commit:
fix CI

Commit:
fix CI

Commit:
fix CI

mrshiposha committed:
fix: bdk xcm jsonnet files

Commit:
fix CI

Commit:
fix CI

Commit:
fix CI

CertainLach commented on .docker/Dockerfile-unique-release:
```diff
@@ -8,8 +8,8 @@ ARG CODE_HASH
-    cd unique-chain && git switch "${CODE_HASH}" --detach && cargo update && \
-    CARGO_INCREMENTAL=0 cargo build --release --features="${FEATURES}" --locked && \
+    cd unique-chain && git switch "${CODE_HASH}" --detach && \
+    CARGO_INCREMENTAL=0 cargo build --release --features="${FEATURES}" --offline && \
```
Offline предполагает что все пакеты уже скачаны и находятся в кеше, нам это не подходит
И даже если бы подходило, --locked нам всё ещё надо

mrshiposha replied:
Обсуждали на мите. Сделаем сначала `cargo fetch`, а потом билд с `--frozen`

CertainLach replied:
Если коротко, то зачем?)

CertainLach replied:
Особенно если cargo fetch не передавать --locked

mrshiposha replied:
С --locked для fetch.

>Если коротко, то зачем?)

Ну, мы же хотим предыдущий релиз таким, каким он был когда был актуален, без изменений. И с той же версией компилятора. А некоторые из внутренних зависимостей используют ещё более глубокие зависимость, разрешая minor update'ы при resolution'е.

Парочка таких глубоких зависимостей хочет edition2024, на которое ругается старый компилятор для старого релиза.
Чтобы не изменять релиз и собрать как есть, кажется, что стоит взять зависимости как есть в Cargo.lock и собрать `--frozen`

CertainLach replied:
Но ведь --locked делает то же самое?..

CertainLach replied:
Он берёт зависимости согласно локфайлу, и запрещает этот локфайл трогать если он устарел

CertainLach replied:
cargo fetch сейчас вызывается без locked, что позволяет ему ставить более новые зависимости чем в локфайле стоят

mrshiposha replied:
>Но ведь --locked делает то же самое?..

Он дополнительно ругается, что Cargo.lock хочется обновить и отказывается собирать.
Это мы словили на практике.

Да и в доках так же написано:
```
--locked
    Asserts that the exact same dependencies and versions are used as when the existing Cargo.lock file was originally generated. Cargo will exit with an error when either of the following scenarios arises:

        * The lock file is missing.
        * Cargo attempted to change the lock file due to a different dependency resolution.

    It may be used in environments where deterministic builds are desired, such as in CI pipelines.
```

CertainLach replied:
--frozen емнип вовсе не то что мы хотим, но я забыл где для него нормальное описание есть

Короче с frozen не будут проверяться гит зависимости несколько я помню

CertainLach replied:
Не помню где это всё было написано, но помню что из-за frozen в nixpkgs для buildCargoPackage сделаны гитовые костыли

CertainLach replied:
Кмк тут изначально оно было правильно сделано просто с cargo build --locked, и я не понимаю чего вы хотите добиться с другими флагами/дёрганьем cargo fetch, и прочим)

CertainLach replied:
> Он дополнительно ругается, что Cargo.lock хочется обновить и отказывается собирать.

Так нам это и надо, мы не хотим чтобы оно в CI пыталось резолюцию провести

Когда оно ругается надо чтобы кто-то явно обновил локфайл и закоммитил его

CertainLach replied:
Для воспроизводимости

> Ну, мы же хотим предыдущий релиз таким, каким он был когда был актуален, без изменений. И с той же версией компилятора. А некоторые из внутренних зависимостей используют ещё более глубокие зависимость, разрешая minor update'ы при resolution'е.

Локфайл в репе должен работать без необходимости что-то при сборке резолвить

Commit:
fix CI

Commit:
fix CI
